validator/validator.go

```diff
@@ -150,7 +150,7 @@ func (v *Validator) ValidateToken(ctx context.Context, tokenString string) (inte
 	}
 
 	if v.customClaims != nil {
-		validatedClaims.CustomClaims = claimDest[1].(CustomClaims)
+		validatedClaims.CustomClaims, _ = claimDest[1].(CustomClaims)
```

Comment:
This isn't actually necessary here, as we know for sure that `v.customClaims` is of type `func() CustomClaims` and on L117 we actually append that to `claimDest` if it's set. So we know for sure that on line 153, `claimDest[1]` is of type `CustomClaims`.

There is however one scenario where we could pass incorrectly a custom claim like this:

```go
	customClaims := func() validator.CustomClaims {
		return nil
	}
```

So what instead we could do is on L116 ensure we don't just check for `if v.customClaims != nil` but also `if v.customClaims() != nil`. 

Wdyt?

Reply:
@sergiught Sorry for the late reply 🙏🏼 

> So what instead we could do is on L116 ensure we don't just check for if v.customClaims != nil but also if v.customClaims() != nil.

I agree with your idea.
I would add this nil guard on L.116, and then in L.153, I would add a code comment to refer to L.116 👍🏼